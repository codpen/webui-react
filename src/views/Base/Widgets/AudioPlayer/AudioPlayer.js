import React, {useState} from "react";
import {Button, Modal} from "reactstrap";
import * as ReactDOM from "react-dom";
import {MODAL_ROOT_ELEMENT} from "../../../../utils/Constants";
import * as PropTypes from "prop-types";
import ErrorBoundary from "../../../../ErrorHandling/ErrorBoundary";

function AudioPlayer({playbackURL, MimeType}) {

    const [preview, setPreview] = useState(true);

    function hideFull(e) {
        e.stopPropagation();
        setPreview(!preview);

    }

    let element;
    if (preview) {
        element = (
            <div className="img-thumbnail w-100 text-center" data-test="audioPlayerWidget">
                <Button color="link" onClick={hideFull}>
                    {/*<i className="fa fa-play-circle fa-4x"/>*/}
                    <audio controls>
                        <source src={playbackURL} type={MimeType}/>
                        Your browser does not support the audio tag.
                    </audio>
                </Button>
            </div>
        )
    } else {

        // Load the video


        element = ReactDOM.createPortal((
            <Modal className="task-modal d-none d-sm-block" data-test="videoPlayerWidget" isOpen={!preview}
                   toggle={hideFull}>


                <audio controls>
                    <source src={playbackURL} type={MimeType}/>
                    Your browser does not support the audio tag.
                </audio>

            </Modal>
        ), document.getElementById(MODAL_ROOT_ELEMENT));
    }

    return (
        <ErrorBoundary>
            {element}
        </ErrorBoundary>
    )


}

AudioPlayer.propTypes = {
    playbackURL: PropTypes.string.isRequired,
    MimeType: PropTypes.string.isRequired
};

export default AudioPlayer;