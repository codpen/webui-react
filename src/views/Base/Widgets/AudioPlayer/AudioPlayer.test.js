import React from "react";
import {shallow} from "enzyme";
import {findByTestAttr, testStore} from "../../../../../Utils";
import toJson from "enzyme-to-json";
import AudioPlayer from "./AudioPlayer";

const setUp = (intialState = {}, props = {}) => {
    const store = testStore(intialState);
    const component = shallow(<AudioPlayer {...props} store={store}/>);
    return component;
};


describe('Audio Player Widget', function () {


    describe('renders', function () {
        let wrapper;
        beforeEach(() => {
            const initialState = {};


            const props = {
                playbackURL: '',
                MimeType: ''
            };
            wrapper = setUp(initialState, props)
        });

        it('should render without crashing', function () {
            const component = findByTestAttr(wrapper, "audioPlayerWidget");
            expect(component).toHaveLength(1);
        });

        it('should match snapshot', function () {
            expect(toJson(wrapper)).toMatchSnapshot()
        });

    });

});