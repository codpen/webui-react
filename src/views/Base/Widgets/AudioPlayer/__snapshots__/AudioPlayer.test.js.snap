// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Audio Player Widget renders should match snapshot 1`] = `
<ErrorBoundary>
  <div
    className="img-thumbnail w-100 text-center"
    data-test="audioPlayerWidget"
  >
    <Button
      color="link"
      onClick={[Function]}
      tag="button"
    >
      <audio
        controls={true}
      >
        <source
          src=""
          type=""
        />
        Your browser does not support the audio tag.
      </audio>
    </Button>
  </div>
</ErrorBoundary>
`;
